lib/cmdlib/instance_migration.py

```diff
@@ -916,6 +917,10 @@ def _ExecMigration(self):
 
     self.feedback_fn("* starting memory transfer")
     last_feedback = time.time()
+
+    migration_caps = \
+      self.instance.hvparams.get(constants.HV_KVM_MIGRATION_CAPS, None)
```

Comment:
If you return [] instead of None in the get() function when we can't find the KVM migration caps, you don't need to test if the list is empty in the following postcopy_enabled = ... statement.

You can just do

migration_caps = self.instance.hvparams.get(constants.HV_KVM_MIGRATION_CAPS, [])
postcopy_enabled = 'postcopy-ram' in migration_caps

if it were None, it would except obviously so we prevent that here.

This has the advantage that postcopy_enabled will always be a boolean, otherwise in the situation of an empty list it would end up being [], which tests for false but is a different type and I don't like having the type of a variable change based on the context.